components/cli/pkg/commands/image/delete_image.go

```diff
@@ -16,50 +16,54 @@
  * under the License.
  */
 
-package commands
+package image
 
 import (
-	"os"
+	"fmt"
 	"path"
 	"regexp"
 
-	"github.com/cellery-io/sdk/components/cli/pkg/image"
-
+	"github.com/cellery-io/sdk/components/cli/cli"
 	"github.com/cellery-io/sdk/components/cli/pkg/constants"
-	"github.com/cellery-io/sdk/components/cli/pkg/util"
+	"github.com/cellery-io/sdk/components/cli/pkg/image"
 )
 
-func RunDeleteImage(images []string, regex string, deleteAll bool) {
-	imagesInRepo := getImagesArray()
+func RunDeleteImage(cli cli.Cli, images []string, regex string, deleteAll bool) error {
+	repoLocation := cli.FileSystem().Repository()
+	imagesInRepo, err := getImagesArray(cli)
+	if err != nil {
+		return fmt.Errorf("error getting images array, %v", err)
+	}
 	for _, imageInRepo := range imagesInRepo {
 		parsedCellImage, err := image.ParseImageTag(imageInRepo.name)
 		if err != nil {
-			util.ExitWithErrorMessage("Error occurred while parsing cell image", err)
+			return fmt.Errorf("error occurred while parsing cell image, %v", err)
 		}
-		cellImagePath := path.Join(util.UserHomeDir(), constants.CELLERY_HOME, "repo", parsedCellImage.Organization,
+		cellImagePath := path.Join(repoLocation, parsedCellImage.Organization,
 			parsedCellImage.ImageName, parsedCellImage.ImageVersion, parsedCellImage.ImageName+constants.CELL_IMAGE_EXT)
 		if deleteAll {
-			_ = os.RemoveAll(cellImagePath)
+			return cli.FileSystem().RemoveAll(cellImagePath)
 		} else {
 			if regex != "" {
 				// Check if image name matches regex pattern
 				regexMatches, err := regexp.MatchString(regex, imageInRepo.name)
 				if err != nil {
-					util.ExitWithErrorMessage("Error checking if pattern matches with image name", err)
+					return fmt.Errorf("error checking if pattern matches with image name, %v", err)
 				}
 				if regexMatches {
-					_ = os.RemoveAll(cellImagePath)
+					return cli.FileSystem().RemoveAll(cellImagePath)
 					continue
```

Comment:
We don't need to continue here if we are returning

Reply:
Updated. Checked if RemoveAll returns an error and returned only if there is an error. continue statement is needed.